src/main/java/gregtech/api/recipes/Recipe.java

```diff
@@ -77,41 +78,77 @@ public Recipe(List<CountableIngredient> inputs, List<ItemStack> outputs, List<Ch
         this.inputs.sort(Comparator.comparing(CountableIngredient::getCount).reversed());
     }
 
+    public final boolean matches(boolean consumeIfSuccessful, IItemHandlerModifiable inputs, IMultipleTankHandler fluidInputs, MatchingMode matchingMode) {
+        return matches(consumeIfSuccessful, GTUtility.itemHandlerToList(inputs), GTUtility.fluidHandlerToList(fluidInputs), matchingMode);
+    }
+
     public final boolean matches(boolean consumeIfSuccessful, IItemHandlerModifiable inputs, IMultipleTankHandler fluidInputs) {
-        return matches(consumeIfSuccessful, GTUtility.itemHandlerToList(inputs), GTUtility.fluidHandlerToList(fluidInputs));
+        return matches(consumeIfSuccessful, GTUtility.itemHandlerToList(inputs), GTUtility.fluidHandlerToList(fluidInputs), MatchingMode.DEFAULT);
     }
 
     public boolean matches(boolean consumeIfSuccessful, List<ItemStack> inputs, List<FluidStack> fluidInputs) {
-        int[] fluidAmountInTank = new int[fluidInputs.size()];
-        int[] itemAmountInSlot = new int[inputs.size()];
+        return matches(consumeIfSuccessful, inputs, fluidInputs, MatchingMode.DEFAULT);
+    }
 
-        for (int i = 0; i < fluidAmountInTank.length; i++) {
-            FluidStack fluidInTank = fluidInputs.get(i);
-            fluidAmountInTank[i] = fluidInTank == null ? 0 : fluidInTank.amount;
+    /**
+     * This methods aim to verify if the current recipe matches the given inputs according to matchingMode mode.
+     *
+     * @param consumeIfSuccessful if true and matchingMode is equal to {@link MatchingMode#DEFAULT} will consume the inputs of the recipe.
+     * @param inputs              Items input or Collections.emptyList() if none.
+     * @param fluidInputs         Fluids input or Collections.emptyList() if none.
+     * @param matchingMode        How this method should check if inputs matches according to {@link MatchingMode} description.
+     * @return true if the recipe matches the given inputs false otherwise.
+     */
+    public boolean matches(boolean consumeIfSuccessful, List<ItemStack> inputs, List<FluidStack> fluidInputs, MatchingMode matchingMode) {
+
+        if (matchingMode == MatchingMode.IGNORE_FLUIDS && getInputs().isEmpty()) {
+            return false;
         }
-        for (int i = 0; i < itemAmountInSlot.length; i++) {
-            ItemStack itemInSlot = inputs.get(i);
-            itemAmountInSlot[i] = itemInSlot.isEmpty() ? 0 : itemInSlot.getCount();
+        if (matchingMode == MatchingMode.IGNORE_ITEMS && getFluidInputs().isEmpty()) {
+            return false;
         }
 
-        for (FluidStack fluid : this.fluidInputs) {
-            int fluidAmount = fluid.amount;
-            boolean isNotConsumed = false;
-            if (fluidAmount == 0) {
-                fluidAmount = 1;
-                isNotConsumed = true;
+        Pair<Boolean, Integer[]> fluids = matchesFluid(fluidInputs);
+        if (!fluids.getKey()) {
+            return false;
+        }
```

Comment:
This construct should not be called when `matchingMode == MatchingMode.IGNORE_FLUIDS` as it is not needed and it will return incorrect value and recipe would not be correctly matched. You can test this on Tungstate Dust in Electrolyzer.